File: baekjoon/10828-스택/제문/10828.kt
package baekjoon

import java.util.*

fun main() = with(System.`in`.bufferedReader()) {
    var caseNum = readLine().toInt()
    val stack = Stack<Int>()

    while (caseNum-- > 0) {
        val commands = readLine().split(" ")

        when (commands[0]) {
            "push" -> stack.push(commands[1].toInt())
            "pop" -> println(if (stack.isNotEmpty()) stack.pop() else -1)
            "size" -> println(stack.size)
            "empty" -> println(if (stack.isEmpty()) 1 else 0)
            "top" -> println(if (stack.isNotEmpty()) stack.peek() else -1)
        }
    }
}

File: baekjoon/2주차/10828-스택/우섭/10828.py
import sys

n = int(sys.stdin.readline())
stack = []
for _ in range(n):
    input = sys.stdin.readline()
    if 'push' in input:
        stack.append(int(input.split()[1]))
    elif 'pop' in input:
        print(stack.pop() if len(stack) > 0 else -1)
    elif 'size' in input:
        print(len(stack))
    elif 'empty' in input:
        print(1 if len(stack) == 0 else 0)
    elif 'top' in input:
        print(stack[-1] if len(stack) != 0 else -1)